### PR TITLE
feat: store generated changeset into ingestion log

### PR DESCRIPTION
**Pull request by leoparente**
image

### CODE REVIEW

File: diode-server/reconciler/ingestion_processor.go [hunk 1]
```diff
@@ -356,6 +376,24 @@ func normalizeIngestionLog(l []byte) []byte {
...
+	var brotliBuf bytes.Buffer
+	brotliWriter := brotli.NewWriter(&brotliBuf)
+	if _, err = brotliWriter.Write(csJSON); err != nil {
+		return "", fmt.Errorf("failed to compress changeset: %v", err)
+	}
+	if err = brotliWriter.Close(); err != nil {
+		return "", fmt.Errorf("failed to compress changeset: %v", err)
+	}
+
+	return base64.StdEncoding.EncodeToString(brotliBuf.Bytes()), nil
```
Comment (mfiedorowicz) [inline diode-server/reconciler/ingestion_processor.go]: Did we check the size of raw brotli bytes output vs brotli bytes encoded with base64?

File: diode-proto/diode/v1/reconciler.proto [hunk 2]
```diff
@@ -70,6 +70,12 @@ message IngestionMetrics {
   int32 no_changes = 5;
 }
 
+// A change set
+message ChangeSet {
+  string id = 1; // A change set ID
+  string data = 2; // Base64 string representing the change set
```
Comment (mfiedorowicz) [inline diode-proto/diode/v1/reconciler.proto]: Could we ament comment for `data`, please? I'd suggest to use something generic, as data can be of any string form (raw JSON string or brotli or something else), e.g.
```suggestion
  string data = 2; // A change set data
```

File: diode-server/reconciler/ingestion_processor_internal_test.go
Comment (mfiedorowicz) [inline diode-server/reconciler/ingestion_processor_internal_test.go:1]: Can we add tests for compressed change sets, please? E.g. to assert compression works as expected with `compressChangeSet` function.

Comment (leoparente) [reply diode-server/reconciler/ingestion_processor_internal_test.go:1]: added it

**Review by mfiedorowicz — APPROVED** [review]
LGTM

**Pull request merged**